### PR TITLE
cleanup record before save

### LINKED ISSUE #144: Delete empty elements on save
After https://github.com/inveniosoftware-contrib/ng2-json-editor/pull/318 has been merged there might still be empty elements in the final record.

Add a clean up function that on Save will remove all empty strings, arrays and objects.

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds cleanup service.

* Fixes #144.

* Upgrades ng2-json-editor.

### CODE REVIEW

File: src/app/shared/services/record-cleanup.service.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            this.cleanup(value[key]);
+          }
+          if (this.isEmpty(value[key])) {
+            delete value[key];
+          }
+        });
+    }
+  }
+
+  private isEmpty(value: any) {
```
Comment (zzacharo) [inline src/app/shared/services/record-cleanup.service.ts]: Why not using _.isEmpty() of lodash?

File: src/app/shared/services/record-cleanup.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      nestedEmpty: { foo: { nested: { evenMore: '' } } },
+      undefinedProperty: undefined,
+      prop1: '',
+      prop2: 0,
+      prop3: 1,
+      prop4: { foo: 'bar', empty: [] },
+      prop5: [1, undefined]
+    };
+    let expected = {
+      prop1: '',
```
Comment (jmartinm) [inline src/app/shared/services/record-cleanup.service.spec.ts]: This should not be left.

Adding a `expect` would be useful in this test ;)

File: src/app/shared/services/record-cleanup.service.ts (hunk 1 above)
Comment (jmartinm) [inline src/app/shared/services/record-cleanup.service.ts]: @harunurhan you could check this.

Comment (harunurhan) [inline src/app/shared/services/record-cleanup.service.ts]: we have slightly different logic to decide if empty or not, I think.

**Pull request merged**